test/suites/base.bash

```diff
@@ -964,22 +964,60 @@ EOF
     umask $saved_umask
 
     # -------------------------------------------------------------------------
-    TEST "No object file"
+    TEST "No object file, prefixed"
 
-    cat <<'EOF' >test_no_obj.c
-int test_no_obj;
-EOF
-    cat <<'EOF' >prefix-remove.sh
+    cat <<'EOF' >prefix-noop.sh
 #!/bin/sh
-"$@"
-[ x$2 = x-fcolor-diagnostics ] && shift
-[ x$2 = x-fdiagnostics-color ] && shift
-[ x$2 = x-std=gnu99 ] && shift
-[ x$3 = x-o ] && rm $4
```

Comment:
Not sure what these lines do, but I had to remove them to support gcc 4.8.5

Reply:
`-std=gnu99` was added in 1be74015e3970d4679beab8b8beb808bde6ff4c6. Some users had set `CC="gcc -std=gnu99"` in order to be able to compile ccache. `-fcolor-diagnostics` and `-fdiagnostics-color` were added in 396df7efdf26a, presumably for similar reasons.

Not running the compiler at all in the prefix program, like you suggest, indeed sounds better.